### ISSUE TITLE
Feature request: configurable spaces around infix types

### ISSUE THREAD

**Issue by Atry** [olafurpg/scalafmt#660]
I created a compiler plugin to provide `##` syntax https://github.com/ThoughtWorksInc/implicit-dependent-type
I hope people feel that `Foo[Bar]##Baz` is similar to native syntax `Foo[Bar]#Baz`. However, there's currently no configuration to do that, and scalafmt always add spaces around `##`, i.e. `Foo[Bar] ## Baz`.

- **Version**: 0.5
- **Integration**: IntelliJ
- **Configuration**: default
## Unexpected formatting output

Obtained, scalafmt formats code like this:
```scala
val hlistForFoo: Generic[Foo] ## Repr = 1 :: "xxx" :: HNil
```

Expected, I would like the formatted output to look like this:
```scala
val hlistForFoo: Generic[Foo]##Repr = 1 :: "xxx" :: HNil
```

**Comment by olafurpg** [comment]
As discussed on gitter, I am happy to add an option to configure the space around infix type parameters. I suggest to name the flag `spaces.neverAroundInfixTypes: Seq[String] = []`, which you could configure to be `["##"]`. 

If someone is motivated to take on this issue, the tutorial should contain all the necessary instructions to fix it, see https://github.com/olafurpg/scalafmt/blob/master/Tutorial.md

**Comment by olafurpg** [comment]
This was fixed by @ysusuk! In 0.5.2 you will be able to configure
```
spaces.neverAroundInfixTypes = ["##"]
```
